beacon_node/network/src/sync/sampling.rs

```diff
@@ -0,0 +1,486 @@
+use self::request::ActiveColumnSampleRequest;
+
+use super::network_context::{LookupFailure, SyncNetworkContext};
+use beacon_chain::BeaconChainTypes;
+use fnv::FnvHashMap;
+use lighthouse_network::{PeerAction, PeerId};
+use rand::{seq::SliceRandom, thread_rng};
+use slog::{debug, error, warn};
+use std::{
+    collections::hash_map::Entry, collections::HashMap, marker::PhantomData, sync::Arc,
+    time::Duration,
+};
+use types::{data_column_sidecar::ColumnIndex, DataColumnSidecar, Hash256, Slot};
+
+pub type SamplingResult = Result<(), SamplingError>;
+
+#[derive(Debug, Hash, PartialEq, Eq, Clone, Copy)]
+pub struct SamplingId {
+    pub id: SamplingRequester,
+    pub column_index: ColumnIndex,
+}
+
+#[derive(Debug, Hash, PartialEq, Eq, Clone, Copy)]
+pub enum SamplingRequester {
+    ImportedBlock(Hash256),
```

Comment:
Does this mean we're only triggering sampling after importing a block? What other variants are you thinking to add?